sdk/src/Core/Amazon.Util/AWSSDKUtils.cs

```diff
@@ -1761,4 +1803,15 @@ public void Reset()
             _count = 0;
         }
     }
+    public class UriComponent
```

Comment:
Could this be a readonly struct or a record struct?

Reply:
I believe i could make it a readonly struct but is there any reason to?